### PR TITLE
[Compilation][MSVC][C++17] Fix error during compilation due to deprecated warnings

### LINKED ISSUE #10137: [Compilation][Windows][C++17] Error during compilation
**Description**

~~~
C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\VC\Tools\MSVC\14.29.30133\include\xutility(1169): warning C4996: 'std::iterator<IC,T,ptrdiff_t,T *,T &>::iterator_category': warning STL4015: The std::iterator class template (used as a base class to provide typedefs) is deprecated in C++17. (The <iterator> header is NOT deprecated.) The C++ Standard has never required user-defined iterators to derive from std::iterator. To fix this warning, stop deriving from std::iterator and start providing publicly accessible typedefs named iterator_category, value_type, difference_type, pointer, and reference. Note that value_type is required to be non-const, even for constant iterators. You can define _SILENCE_CXX17_ITERATOR_BASE_CLASS_DEPRECATION_WARNING or _SILENCE_ALL_CXX17_DEPRECATION_WARNINGS to acknowledge that you have received this warning.
~~~

**Scope**

- Compilation
- KratosCore

**To Reproduce**

Windows 10. MSVSC 2019.

**Expected behavior**
To compile

**Environment**

- OS: Windows 10
- Branch: master
- MSVSC 2019.

### PR DESCRIPTION
**Pull request by loumalouomega**
**📝 Description**

Fixes #10137

**🆕 Changelog**

- [Fix error during compilation](https://github.com/KratosMultiphysics/Kratos/commit/b571a0d3b8f186bc1d76ce4e956a8c0f5de8069b)

### CODE REVIEW

**Review by roigcarlo — CHANGES_REQUESTED** [review]
If you want to add this flag put it optional, we cannot hide all warnings about C++17

**Comment by loumalouomega** [comment]
> If you want to add this flag put it optional, we cannot hide all warnings about C++17

Okidoki

File: CMakeLists.txt [hunk 1]
```diff
@@ -244,6 +244,8 @@ if(${MSVC})
   SET (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /W1 /bigobj /EHsc /Zc:__cplusplus -DBOOST_ALL_NO_LIB -D_SCL_SECURE_NO_WARNINGS")
   string( REPLACE "/W3" "" CMAKE_C_FLAGS ${CMAKE_C_FLAGS} )
   string( REPLACE "/W3" "" CMAKE_CXX_FLAGS ${CMAKE_CXX_FLAGS} )
+  # Remove deprecated warnings of C++17
+  add_compile_definitions(_SILENCE_ALL_CXX17_DEPRECATION_WARNINGS)
```
Comment (loumalouomega) [inline CMakeLists.txt]:
```suggestion
  if(${_SILENCE_ALL_CXX17_DEPRECATION_WARNINGS})
    add_compile_definitions(_SILENCE_ALL_CXX17_DEPRECATION_WARNINGS)
  endif(${_SILENCE_ALL_CXX17_DEPRECATION_WARNINGS})
```

**Comment by roigcarlo** [comment]
Not but I mean, just add `-D_SILENCE_ALL_CXX17_DEPRECATION_WARNINGS` in the `CMAKE_CXX_FLAGS`, no need to add another Kratos option for that.

**Comment by loumalouomega** [comment]
> Not but I mean, just add `-D_SILENCE_ALL_CXX17_DEPRECATION_WARNINGS` in the `CMAKE_CXX_FLAGS`, no need to add another Kratos option for that.

Huuuuuum, you are right

**Pull request closed without merging**